src/tests/ftest/control/dmg_config_generate.py

```diff
@@ -0,0 +1,256 @@
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_1
+        """
+        self.run_test(None, None, None)
+
+    def test_dmg_config_generate_2(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_2
+        """
+        self.run_test(None, 1, None)
+
+    def test_dmg_config_generate_3(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_3
+        """
+        self.run_test(None, 2, None)
+
+    def test_dmg_config_generate_4(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_4
+        """
+        self.run_test(1, None, None)
+
+    def test_dmg_config_generate_5(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_4
+        """
+        self.run_test(2, None, None)
+
+    def test_dmg_config_generate_6(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_6
+        """
+        self.run_test(1, 1, "ethernet")
+
+    def test_dmg_config_generate_7(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_7
+        """
+        self.run_test(2, 2, "infiniband")
+
+    def test_dmg_config_generate_8(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        We expect this test to fail since requested storage is not available.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_8
+        """
+        try:
+            self.run_test(10, 10, "infiniband")
+        except CommandFailure as error:
+            self.log.info("Expected failure: {}".format(error))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments